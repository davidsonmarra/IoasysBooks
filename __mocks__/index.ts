import BookDTO from '../src/@types/BookDTO';

interface IMockFetchDataBooks {
  data: BookDTO[];
  page: number;
  totalPages: number;
  totalItems: number;
}

export const mockBook: BookDTO = {
  id: '1',
  title: 'Test Title',
  imageUrl: 'https://www.test.com',
  authors: ['Test Author'],
  pageCount: 100,
  publisher: 'Test Publisher',
  published: 1980,
  category: 'Test Category',
  language: 'Test Language',
  isbn10: 'Test ISBN10',
  isbn13: 'Test ISBN13',
  description: 'Test Description'
};

export const mockPreloadedStateEmpty = {
  profile: {
    email: '',
    token: '',
    error: {} as Error,
    isLogged: false,
    isLoading: false
  },
  books: {
    isEnd: false,
    loadingFetchBooks: false,
    error: {} as Error,
    booksData: [],
    category: { key: '', title: '' },
    search: ''
  }
};

export const mockPreloadedStateFilled = {
  profile: {
    email: 'davidson',
    token: 'teste',
    error: {} as Error,
    isLogged: true,
    isLoading: false
  },
  books: {
    isEnd: false,
    loadingFetchBooks: false,
    error: {} as Error,
    booksData: [mockBook],
    category: { key: '', title: '' },
    search: ''
  }
};

export const mockFetchDataBooks: IMockFetchDataBooks = {
  data: [
    {
      ...mockBook,
      id: '2',
      title: 'Harry Potter 1'
    },
    {
      ...mockBook,
      id: '3',
      title: 'Harry Potter 2'
    },
    {
      ...mockBook,
      id: '4',
      title: 'Harry Potter 3'
    },
    {
      ...mockBook,
      id: '5',
      title: 'Harry Potter 4'
    },
    {
      ...mockBook,
      id: '6',
      title: 'Harry Potter 5'
    }
  ],
  page: 1,
  totalPages: 1,
  totalItems: 2
};
