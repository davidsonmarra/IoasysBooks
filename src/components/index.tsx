export { Input } from './form/Input';
export { SearchInput } from './form/SearchInput';
export { FilterModal } from './FilterModal';
export { CardBook } from './CardBook';
export { ShimmerCardBook } from './ShimmerCardBook';
