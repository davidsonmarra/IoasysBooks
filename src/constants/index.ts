const constants = {
  BASE_URL: 'https://books.ioasys.com.br/api/v1',
  TIMEOUT: 8000,
  asyncStorageUserKey: '@books_asyncStorageUserKey',
  asyncStorageUserId: '@books_asyncStorageUserId',
  asyncStorageUserRefresh: '@books_asyncStorageUserRefresh'
};

export default constants;
