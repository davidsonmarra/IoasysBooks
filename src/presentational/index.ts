export { SignIn } from './SignIn';
export { BookDetails } from './BookDetails';
export { Home } from './Home';
