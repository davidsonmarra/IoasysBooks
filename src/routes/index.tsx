import React from 'react';
import { NavigationContainer } from '@react-navigation/native';
import { PublicRoutes } from './stacks/public.routes';
import { useSelector } from 'react-redux';
import { IRootState } from '../store/store';
import { AuthRoutes } from './stacks/auth.routes';

export function Routes() {
  const { isLogged } = useSelector(({ profile }: IRootState) => profile);

  return <NavigationContainer>{isLogged ? <AuthRoutes /> : <PublicRoutes />}</NavigationContainer>;
}
