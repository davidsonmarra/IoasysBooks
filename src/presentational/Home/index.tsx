import React, { useEffect, useState } from 'react';
import { FlatListProps, FlatList, Keyboard, RefreshControl } from 'react-native';
import styled, { useTheme } from 'styled-components/native';
import LogoSvg from '../../assets/logo-dark.svg';
import FilterSvg from '../../assets/filter.svg';
import TitleSvg from '../../assets/title-dark.svg';
import LogoutSvg from '../../assets/logout.svg';
import { RFValue } from 'react-native-responsive-fontsize';
import BookDTO from '../../@types/BookDTO';
import { CardBook, FilterModal, SearchInput, ShimmerCardBook } from '../../components';
import { FETCH_BOOKS, RESET_BOOKS } from '../../store/slices/booksSlice';
import { useDispatch, useSelector } from 'react-redux';
import { IRootState } from '../../store/store';
import { LOGOUT } from '../../store/slices/profileSlice';
import { NavigationProp, useNavigation } from '@react-navigation/native';
import AuthRootStackParamList from '../../@types/AuthRootStackParamList';
import { useForm } from 'react-hook-form';
import { yupResolver } from '@hookform/resolvers/yup';
import { schema } from './schema';

export interface IFormSearch {
  search: string;
}

export function Home() {
  const { colors } = useTheme();
  const [filterModalVisible, setFilterModalVisible] = useState(false);
  const [offset, setOffset] = useState(1);
  const { navigate } = useNavigation<NavigationProp<AuthRootStackParamList, 'Home'>>();
  const dispatch = useDispatch();
  const { isEnd, booksData, loadingFetchBooks, category } = useSelector(
    ({ books }: IRootState) => books
  );
  const { control, getValues, handleSubmit, reset } = useForm<IFormSearch>({
    resolver: yupResolver(schema),
    mode: 'onSubmit'
  });

  const handleFilterModal = () => setFilterModalVisible(prev => !prev);

  const handleLogout = () => {
    dispatch(RESET_BOOKS());
    dispatch(LOGOUT());
  };

  const onReset = () => {
    setOffset(1);
    reset();
    dispatch(RESET_BOOKS());
    dispatch(FETCH_BOOKS({ offset: 1, category, search: '' }));
  };

  const onSubmit = (form: IFormSearch) => {
    Keyboard.dismiss();
    setOffset(1);
    dispatch(RESET_BOOKS());
    dispatch(FETCH_BOOKS({ offset: 1, category, search: form.search }));
  };

  const onEndReached = () => {
    if (!isEnd) {
      dispatch(FETCH_BOOKS({ offset: offset + 1, category, search: getValues('search') || '' }));
      setOffset(offset + 1);
    }
  };

  const renderListFooterComponent = () =>
    loadingFetchBooks ? (
      <>
        <ShimmerCardBook />
        <ShimmerCardBook />
        <ShimmerCardBook />
      </>
    ) : (
      <StyledFooter />
    );

  function handleGoToBookDetails(book: BookDTO) {
    navigate('BookDetailsScreen', { book });
  }

  const renderItem = ({ item }: { item: BookDTO }) => (
    <CardBook data={item} handleGoToBookDetails={book => handleGoToBookDetails(book)} />
  );

  const refreshControl = (
    <RefreshControl
      colors={[colors.button]}
      tintColor={colors.button}
      refreshing={loadingFetchBooks}
      onRefresh={() => onSubmit({ search: getValues('search') || '' })}
    />
  );

  useEffect(() => {
    dispatch(FETCH_BOOKS({ offset, category, search: getValues('search') || '' }));
  }, []);

  return (
    <StyledContainer>
      <StyledHeader>
        <StyledTitle>
          <StyledLogo />
          <TitleSvg />
        </StyledTitle>
        <StyledLogoutButton accessibilityLabel='logout' onPress={handleLogout}>
          <LogoutSvg />
        </StyledLogoutButton>
      </StyledHeader>
      <StyledSearch>
        <SearchInput
          control={control}
          name='search'
          placeholder='Procure um livro'
          autoCorrect={false}
          autoCapitalize='none'
          searchValue={getValues('search')}
          onSubmit={handleSubmit(onSubmit)}
          reset={onReset}
        />
        <StyledFilterButton onPress={handleFilterModal}>
          <StyledFilter />
        </StyledFilterButton>
      </StyledSearch>
      <StyledList
        data={booksData}
        refreshControl={refreshControl}
        keyExtractor={item => `${item.id}`}
        refreshing={loadingFetchBooks}
        showsVerticalScrollIndicator={false}
        renderItem={renderItem}
        onEndReached={onEndReached}
        onEndReachedThreshold={0.5}
        scrollEventThrottle={16}
        ListFooterComponent={renderListFooterComponent}
        keyboardDismissMode='on-drag'
      />
      <FilterModal
        visible={filterModalVisible}
        handleModal={handleFilterModal}
        setOffset={setOffset}
        getSearch={() => getValues('search')}
      />
    </StyledContainer>
  );
}

export const StyledContainer = styled.View`
  flex: 1;
  background-color: ${({ theme }) => theme.colors.background};
`;

export const StyledLogo = styled(LogoSvg)`
  margin-right: ${RFValue(16.6)}px;
`;

export const StyledHeader = styled.View`
  flex-direction: row;
  justify-content: space-between;
  align-items: center;
  padding: ${RFValue(42)}px ${RFValue(16)}px 0;
`;

export const StyledTitle = styled.View`
  flex-direction: row;
  justify-content: space-between;
  align-items: center;
`;

export const StyledLogoutButton = styled.TouchableOpacity`
  width: ${RFValue(32)}px;
  height: ${RFValue(32)}px;
  justify-content: center;
  align-items: center;
  border: ${RFValue(1)}px solid ${({ theme }) => theme.colors.input_background};
  border-radius: ${RFValue(16)}px;
`;

export const StyledSearch = styled.View`
  flex-direction: row;
  justify-content: space-between;
  align-items: center;
  margin-top: ${RFValue(32)}px;
  padding: 0 ${RFValue(16)}px ${RFValue(32)}px;
`;

export const StyledFilterButton = styled.TouchableOpacity``;

export const StyledFilter = styled(FilterSvg).attrs({
  width: RFValue(24),
  height: RFValue(24)
})``;

export const StyledList = styled(
  FlatList as new (props: FlatListProps<BookDTO>) => FlatList<BookDTO>
).attrs({
  contentContainerStyle: {
    alignItems: 'center',
    paddingBottom: RFValue(24)
  }
})``;

export const StyledFooter = styled.View`
  height: ${RFValue(100)}px;
`;
