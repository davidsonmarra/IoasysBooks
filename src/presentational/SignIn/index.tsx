import React, { useEffect } from 'react';
import {
  Keyboard,
  KeyboardAvoidingView,
  Platform,
  StatusBar,
  TouchableWithoutFeedback
} from 'react-native';
import styled from 'styled-components/native';
import { yupResolver } from '@hookform/resolvers/yup';
import { useForm } from 'react-hook-form';
import { RFValue } from 'react-native-responsive-fontsize';
import { schema } from './schema';
import { Input } from '../../components';
import TitleSvg from '../../assets/title.svg';
import LogoSvg from '../../assets/logo.svg';
import background from '../../assets/background.png';
import Toast from 'react-native-toast-message';
import Animated, {
  useSharedValue,
  useAnimatedStyle,
  interpolate,
  withTiming,
  Easing,
  Extrapolate
} from 'react-native-reanimated';
import { useDispatch, useSelector } from 'react-redux';
import { LOGIN } from '../../store/slices/profileSlice';
import { IRootState } from '../../store/store';
import { isObjectEmpty } from '../../utils';

export interface IFormSignIn {
  email: string;
  password: string;
}

export function SignIn() {
  const { error } = useSelector(({ profile }: IRootState) => profile);

  const dispatch = useDispatch();
  const animation = useSharedValue(0);

  const animationStyleLogo = useAnimatedStyle(() => ({
    marginRight: interpolate(animation.value, [0, 1], [-16.6, 16.6], Extrapolate.CLAMP),
    left: interpolate(animation.value, [0, 1], [50, 0], Extrapolate.CLAMP),
    opacity: interpolate(animation.value, [0, 1], [0, 1])
  }));

  const animationStyleTitle = useAnimatedStyle(() => ({
    right: interpolate(animation.value, [0, 1], [50, 0], Extrapolate.CLAMP),
    opacity: interpolate(animation.value, [0, 1], [0, 1])
  }));

  const animationStyleForm = useAnimatedStyle(() => ({
    opacity: interpolate(animation.value, [0, 1], [0, 1])
  }));

  const {
    control,
    formState: { errors },
    clearErrors,
    getValues,
    setError,
    handleSubmit
  } = useForm<IFormSignIn>({
    resolver: yupResolver(schema),
    mode: 'onSubmit',
    reValidateMode: 'onChange'
  });

  const onSubmit = (form: IFormSignIn) => {
    Keyboard.dismiss();
    dispatch(LOGIN(form));
  };

  useEffect(() => {
    animation.value = withTiming(1, { duration: 1200, easing: Easing.in(Easing.ease) });
  }, []);

  useEffect(() => {
    isObjectEmpty(errors) &&
      Toast.show({
        type: 'error',
        text1: 'Ops... algo não funcionou corretamente',
        text2: errors.email?.message || errors.password?.message,
        props: {
          style: {
            background: 'red'
          }
        }
      });
  }, [errors]);

  useEffect(() => {
    if (isObjectEmpty(error)) {
      setError('email', new Error('Login inválido. Tente novamente'));
      setError('password', new Error('Login inválido. Tente novamente'));
      Toast.show({
        type: 'error',
        text1: 'Ops... algo não funcionou corretamente',
        text2: errors.email?.message || errors.password?.message,
        props: {
          style: {
            background: 'red'
          }
        }
      });
    }
  }, [error]);

  return (
    <TouchableWithoutFeedback testID='SignInScreen' onPress={Keyboard.dismiss}>
      <KeyboardAvoidingView
        style={{ flex: 1 }}
        behavior={Platform.OS === 'ios' ? 'padding' : 'height'}
      >
        <StyledContainer>
          <StatusBar barStyle='light-content' backgroundColor='transparent' translucent />
          <StyledBackground source={background} />
          <StyledTitle>
            <Animated.View style={[animationStyleLogo]}>
              <LogoSvg />
            </Animated.View>
            <Animated.View style={[animationStyleTitle]}>
              <TitleSvg />
            </Animated.View>
          </StyledTitle>
          <AnimatedForm style={[animationStyleForm]}>
            <Input
              name='email'
              placeholder='Email'
              autoCorrect={false}
              autoCapitalize='none'
              control={control}
              error={errors['email']}
              canClearErrorOnFocus={isObjectEmpty(error)}
              getValue={getValues}
              clearErrors={clearErrors}
              errors={errors}
            />
            <Input
              name='password'
              placeholder='Senha'
              secureTextEntry
              autoCorrect={false}
              autoCapitalize='none'
              control={control}
              error={errors['password']}
              getValue={getValues}
              canClearErrorOnFocus={isObjectEmpty(error)}
              onSubmit={handleSubmit(onSubmit)}
              clearErrors={clearErrors}
              errors={errors}
            />
          </AnimatedForm>
        </StyledContainer>
      </KeyboardAvoidingView>
    </TouchableWithoutFeedback>
  );
}

const StyledContainer = styled.View`
  flex: 1;
  justify-content: center;
  padding: 0 ${RFValue(16)}px;
`;

const StyledBackground = styled.ImageBackground`
  position: absolute;
  top: 0;
  bottom: 0;
  left: 0;
  right: 0;
`;

const StyledTitle = styled.View`
  flex-direction: row;
  align-items: center;
  margin-bottom: ${RFValue(50)}px;
`;

const StyledForm = styled.View``;

const AnimatedForm = Animated.createAnimatedComponent(StyledForm);
