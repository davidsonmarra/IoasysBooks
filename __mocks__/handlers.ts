import { rest } from 'msw';
import { mockFetchDataBooks, mockFetchDataRefresh, mockFetchDataSignIn } from '.';
import constants from '../src/constants';

export const handlers = [
  rest.get(`${constants.BASE_URL}/books`, (req, res, ctx) => {
    const search: string = req.url.searchParams.get('title') || '';
    return search.includes('error')
      ? res(
          ctx.status(401),
          ctx.json({
            errorMessage: `Error 401`
          })
        )
      : res(ctx.json(mockFetchDataBooks));
  }),
  rest.post(`${constants.BASE_URL}/auth/sign-in`, async (req, res, ctx) => {
    const { email } = await req.json();
    return email.includes('error')
      ? res(
          ctx.status(401),
          ctx.json({
            errorMessage: `Error 401`
          })
        )
      : res(ctx.json(mockFetchDataSignIn));
  }),
  rest.post(`${constants.BASE_URL}/auth/refresh-token`, async (req, res, ctx) => {
    const { refreshToken } = await req.json();
    return refreshToken.includes('error')
      ? res(
          ctx.status(401),
          ctx.json({
            errorMessage: `Error 401`
          })
        )
      : res(ctx.json(mockFetchDataRefresh));
  })
];
