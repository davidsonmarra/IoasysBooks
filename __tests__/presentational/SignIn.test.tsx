import { SignIn } from '../../src/presentational';
import { mockPreloadedStateEmpty } from '../../__mocks__';
import {
  matchSnapshotWithProviders,
  fireEvent,
  renderWithCustomProviders,
  act,
  waitFor,
  cleanup
} from '../test-utils';
import { server } from '../jest.setup';
import Toast from 'react-native-toast-message';
import { Platform } from 'react-native';

describe('Sign In', () => {
  it('should match a snapshot', () => {
    matchSnapshotWithProviders(<SignIn />);
  });

  afterEach(() => {
    jest.restoreAllMocks();
    cleanup();
    server.resetHandlers();
  });

  afterAll(() => server.close());

  it('should login work correctly', async () => {
    Object.defineProperty(Platform, 'OS', { get: jest.fn(() => 'android') });
    server.listen();
    const {
      render: { getByTestId, getByText },
      store
    } = renderWithCustomProviders(<SignIn />, mockPreloadedStateEmpty);
    const inputEmail = getByTestId(/input-email/i);
    const inputPassword = getByTestId(/input-password/i);

    fireEvent.changeText(inputEmail, 'desafio@ioasys.com.br');
    fireEvent.changeText(inputPassword, '12341234');

    const submitButton = getByText(/entrar/i);

    expect(store.getState().profile.email).toBeFalsy();
    await waitFor(() => act(() => fireEvent.press(submitButton)));
    expect(store.getState().profile.email).toEqual('desafio@ioasys.com.br');
  });

  it('should login render correctly at api error ', async () => {
    server.listen();
    const {
      render: { getByTestId, getByText },
      store
    } = renderWithCustomProviders(<SignIn />, mockPreloadedStateEmpty);
    const inputEmail = getByTestId(/input-email/i);
    const inputPassword = getByTestId(/input-password/i);

    fireEvent.changeText(inputEmail, 'error@ioasys.com.br');
    fireEvent.changeText(inputPassword, '12341234');

    const submitButton = getByText(/entrar/i);

    expect(store.getState().profile.error).toStrictEqual({});
    await waitFor(() => act(() => fireEvent.press(submitButton)));
    expect(store.getState().profile.error.name).toStrictEqual('AxiosError');
  });

  it('should login render correctly at form error in email field', async () => {
    const mockedShowToast = jest.spyOn(Toast, 'show');

    const {
      render: { getByTestId, getByText }
    } = renderWithCustomProviders(<SignIn />, mockPreloadedStateEmpty);
    const inputEmail = getByTestId(/input-email/i);
    const inputPassword = getByTestId(/input-password/i);

    fireEvent.changeText(inputEmail, 'error@');
    fireEvent.changeText(inputPassword, '12341234');

    const submitButton = getByText(/entrar/i);

    await waitFor(() => act(() => fireEvent.press(submitButton)));
    expect(mockedShowToast).toHaveBeenCalledTimes(1);
  });

  it('should login render correctly at form error in password field', async () => {
    const mockedShowToast = jest.spyOn(Toast, 'show');

    const {
      render: { getByTestId, getByText }
    } = renderWithCustomProviders(<SignIn />, mockPreloadedStateEmpty);
    const inputEmail = getByTestId(/input-email/i);
    const inputPassword = getByTestId(/input-password/i);

    fireEvent.changeText(inputEmail, 'error@ioasys.com.br');
    fireEvent.changeText(inputPassword, '');

    const submitButton = getByText(/entrar/i);

    await waitFor(() => act(() => fireEvent.press(submitButton)));
    expect(mockedShowToast).toHaveBeenCalledTimes(1);
  });
});
