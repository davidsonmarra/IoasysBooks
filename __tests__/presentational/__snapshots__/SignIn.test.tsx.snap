// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sign In should match a snapshot 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onLayout={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      Object {
        "paddingBottom": 0,
      },
    ]
  }
  testID="SignInScreen"
>
  <View
    style={
      Array [
        Object {
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
          "justifyContent": "center",
          "paddingBottom": 0,
          "paddingLeft": 31,
          "paddingRight": 31,
          "paddingTop": 0,
        },
      ]
    }
  >
    <View
      accessibilityIgnoresInvertColors={true}
      style={
        Array [
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
        ]
      }
    >
      <Image
        source={
          Object {
            "testUri": "../../../src/assets/background.png",
          }
        }
        style={
          Array [
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            },
            Object {
              "height": undefined,
              "width": undefined,
            },
            undefined,
          ]
        }
      />
    </View>
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "marginBottom": 98,
          },
        ]
      }
    >
      <View
        animatedStyle={
          Object {
            "value": Object {
              "left": 50,
              "marginRight": -16.6,
              "opacity": 0,
            },
          }
        }
        collapsable={false}
        style={
          Object {
            "left": 50,
            "marginRight": -16.6,
            "opacity": 0,
          }
        }
      >
        <SvgMock />
      </View>
      <View
        animatedStyle={
          Object {
            "value": Object {
              "opacity": 0,
              "right": 50,
            },
          }
        }
        collapsable={false}
        style={
          Object {
            "opacity": 0,
            "right": 50,
          }
        }
      >
        <SvgMock />
      </View>
    </View>
    <View
      animatedStyle={
        Object {
          "value": Object {
            "opacity": 0,
          },
        }
      }
      collapsable={false}
      style={
        Array [
          Object {},
          Object {
            "opacity": 0,
          },
        ]
      }
    >
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        forwardedRef={[Function]}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "#00000052",
            "borderColor": "rgba(0, 0, 0, 0)",
            "borderRadius": 8,
            "borderWidth": 2,
            "flexDirection": "row",
            "height": 118,
            "justifyContent": "flex-end",
            "marginBottom": 31,
            "opacity": 1,
            "width": "100%",
          }
        }
      >
        <TextInput
          autoCapitalize="none"
          autoCorrect={false}
          onBlur={[Function]}
          onChangeText={[Function]}
          onFocus={[Function]}
          style={
            Array [
              Object {
                "alignSelf": "flex-end",
                "color": "#FFFFFF",
                "flexBasis": 0,
                "flexGrow": 1,
                "flexShrink": 1,
                "fontFamily": "Heebo-Regular",
                "fontSize": 31,
                "paddingBottom": 16,
                "paddingLeft": 31,
                "paddingRight": 31,
                "paddingTop": 0,
              },
            ]
          }
          testID="input-email"
        />
        <Text
          animatedStyle={
            Object {
              "value": Object {
                "bottom": 8,
                "fontSize": 16,
                "left": 16,
              },
            }
          }
          collapsable={false}
          style={
            Array [
              Object {
                "color": "#FFFFFF",
                "fontFamily": "Heebo-Regular",
                "fontSize": 31,
                "opacity": 0.5,
                "position": "absolute",
              },
              Object {
                "bottom": 8,
                "fontSize": 16,
                "left": 16,
              },
            ]
          }
        >
          Email
        </Text>
      </View>
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        forwardedRef={[Function]}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "#00000052",
            "borderColor": "rgba(0, 0, 0, 0)",
            "borderRadius": 8,
            "borderWidth": 2,
            "flexDirection": "row",
            "height": 118,
            "justifyContent": "flex-end",
            "marginBottom": 31,
            "opacity": 1,
            "width": "100%",
          }
        }
      >
        <TextInput
          autoCapitalize="none"
          autoCorrect={false}
          onBlur={[Function]}
          onChangeText={[Function]}
          onFocus={[Function]}
          secureTextEntry={true}
          style={
            Array [
              Object {
                "alignSelf": "flex-end",
                "color": "#FFFFFF",
                "flexBasis": 0,
                "flexGrow": 1,
                "flexShrink": 1,
                "fontFamily": "Heebo-Regular",
                "fontSize": 31,
                "paddingBottom": 16,
                "paddingLeft": 31,
                "paddingRight": 31,
                "paddingTop": 0,
              },
            ]
          }
          testID="input-password"
        />
        <View
          accessibilityState={
            Object {
              "disabled": false,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "alignSelf": "center",
              "backgroundColor": "#FFFFFF",
              "borderRadius": 86,
              "height": 78,
              "justifyContent": "center",
              "marginRight": 31,
              "opacity": 1,
              "paddingBottom": 16,
              "paddingLeft": 0,
              "paddingRight": 0,
              "paddingTop": 16,
              "width": 167,
            }
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#B22E6F",
                  "fontFamily": "Heebo-Medium",
                  "fontSize": 31,
                },
              ]
            }
          >
            Entrar
          </Text>
        </View>
        <Text
          animatedStyle={
            Object {
              "value": Object {
                "bottom": 8,
                "fontSize": 16,
                "left": 16,
              },
            }
          }
          collapsable={false}
          style={
            Array [
              Object {
                "color": "#FFFFFF",
                "fontFamily": "Heebo-Regular",
                "fontSize": 31,
                "opacity": 0.5,
                "position": "absolute",
              },
              Object {
                "bottom": 8,
                "fontSize": 16,
                "left": 16,
              },
            ]
          }
        >
          Senha
        </Text>
      </View>
    </View>
  </View>
</View>
`;
